src/test/java/org/jetbrains/bsp/bazel/BazelBspServerTestData.java

```diff
@@ -104,4 +109,16 @@
 
   static final InverseSourcesResult EXPECTED_INVERSE_SOURCES =
       new InverseSourcesResult(ImmutableList.of(ID_2));
+
+  static final ScalaMainClassesParams SCALA_MAIN_CLASSES_PARAMS =
+      new ScalaMainClassesParams(Collections.singletonList(ID_1));
+
+  static final ScalaMainClassesResult EXPECTED_SCALA_MAIN_CLASSES =
+      new ScalaMainClassesResult(
+          Collections.singletonList(
+              new ScalaMainClassesItem(
+                  ID_1,
+                  Collections.singletonList(
+                      new ScalaMainClass(
+                          "example.Example", Collections.emptyList(), Collections.emptyList())))));
```

Comment:
We probably should guarantee that the rest of the Scala Main Class is also working, by providing arguments and jvmOptions to the example class, wdyt?

Reply:
You should have access to a `scalacopts` and I think `jvm_flags` should be substituted by a union of `jvm_flags`, `scalac_jvm_flags` and `javac_jvm_flags`: https://github.com/bazelbuild/rules_scala/blob/773c992c0fcbf0e18355eaef1783c49915455c12/docs/scala_binary.md .